packages/cli/src/ui/components/shared/text-buffer.ts

```diff
@@ -1299,6 +1300,45 @@ export function textBufferReducer(
       return { ...nextState, lines: newLines, preferredCol: null };
     }
 
+    case 'delete_word_backward': {
```

Comment:
how is this different from deleteWordLeft? Please align with that method rather than creating another one. note that that command already has a keyboard shortcut tied to ctrl backspace which just likely wasn't working on your keyboard without the other fixes you landed this pr.

Reply:
Yep, you're right there's a lot of duplication that I merged together. The logic I added improves on the word boundary logic, which I moved into `deleteWordLeft` and removed the `delete_word_backward` function. 

The main difference is that the new logic is much smarter about what it considers a "word." The old version would treat things like `variable_name` or `path/to/file` as a single word. It's also better with unicode and punctuation, so something like `foo!!!` is correctly treated as two words.

Let me know if this improvement is good and if not I can just resort back to the old `deleteWordLeft` logic.